src/spaces/graph.jl

```diff
@@ -68,21 +68,29 @@ function add_agent_to_space!(
     return agent
 end
 
-# The following two is for the discrete space API:
+# The following is for the discrete space API:
 ids_in_position(n::Integer, model::ABM{<:GraphSpace}) = model.space.s[n]
 # NOTICE: The return type of `ids_in_position` must support `length` and `isempty`!
 
 #######################################################################################
 # Neighbors
 #######################################################################################
-function nearby_ids(pos::Int, model::ABM{<:GraphSpace}, args...; kwargs...)
-    np = nearby_positions(pos, model, args...; kwargs...)
-    # This call is faster than reduce(vcat, ..), or Iterators.flatten
-    vcat(model.space.s[pos], model.space.s[np]...)
+function nearby_ids(pos::Int, model::ABM{<:GraphSpace}, r; kwargs...)
```

Comment:
All other changes in the PR look good, a lot of nice simplifications.

I remember checking this `vcat` method quite extensively and it was faster than all the other methods I tried by a good margin. Will be great to see some benchmarks on this new method of yours to see how it compares!